### PR TITLE
Implemented IpTunnel_removeConnection

### PR DESCRIPTION
**Pull request by 4L3Xx**
Check this code.  It seems to be working but I am not sure if anything else needs to be done when removing the connection.

### CODE REVIEW

File: tunnel/IpTunnel.c [hunk 1]
```diff
@@ -105,6 +105,33 @@ static struct IpTunnel_Connection* newConnection(bool isOutgoing, struct IpTunne
...
 
+static void deleteConnection(struct IpTunnel_Connection* conn, struct IpTunnel_pvt* context)
+{
+    // Delete connection and shift the list elements following the removed connection
+
+    int i = (((char *)conn)-((char *)&context->pub.connectionList.connections[0]))
+                                                  / sizeof(struct IpTunnel_Connection);
+
+    // Sanity check
+    if (i < 0 || i >= (signed int)context->pub.connectionList.count) {
```
Comment (cjdelisle) [inline tunnel/IpTunnel.c]: Unless you think there's a good reason why it might be false you should  Assert_true()

File: tunnel/IpTunnel_admin.c [hunk 2]
```diff
@@ -144,14 +144,13 @@ static void removeConnection(Dict* args,
                              struct Allocator* requestAlloc)
 {
     struct Context* context = vcontext;
-/*
+
     int conn = (int) *(Dict_getInt(args, String_CONST("connection")));
-    char* error = "none";
     if (IpTunnel_removeConnection_NOT_FOUND == IpTunnel_removeConnection(conn, context->ipTun)) {
-        error = "not found";
+        sendError("not found", txid, context->admin);
```
Comment (cjdelisle) [inline tunnel/IpTunnel_admin.c]: sending ```not_found``` is most common now (this has evolved somewhat)

**Comment by cjdelisle** [comment]
Looks ok w/ minor changes

File: tunnel/IpTunnel_admin.c (hunk 2 above)
Comment (4L3Xx) [inline tunnel/IpTunnel_admin.c]: Could you clarify?  sendError("not_found", txid, context->admin); ?

Comment (lgierth) [inline tunnel/IpTunnel_admin.c]: Yes that

**Comment by 4L3Xx** [comment]
Ok, I made the fixes.

**Pull request merged**